coalib/results/Diff.py

```diff
@@ -197,11 +198,72 @@ def unified_diff(self):
         Note that the unified diff is not deterministic and thus not suitable
         for equality comparison.
         """
-        return ''.join(difflib.unified_diff(
+        return ''.join(self.coala_unified_diff(
             self.original,
             self.modified,
             tofile=self.rename if isinstance(self.rename, str) else ''))
 
+    def coala_unified_diff(a, b, fromfile='', tofile='', fromfiledate='',
+                           tofiledate='', n=3, lineterm='\n'):
+        r"""
+        Compare two sequences of lines; generate the delta as a unified diff.
+
+        Unified diffs are a compact way of showing line changes and a few
+        lines of context.  The number of context lines is set by 'n' which
+        defaults to three.
+
+        By default, the diff control lines (those with ---, +++, or @@) are
+        created with a trailing newline.  This is helpful so that inputs
+        created from file.readlines() result in diffs that are suitable for
+        file.writelines() since both the inputs and outputs have trailing
+        newlines.
+
+        For inputs that do not have trailing newlines, set the lineterm
+        argument to "" so that the output will be uniformly newline free.
+
+        The unidiff format normally has a header for filenames and modification
+        times.  Any or all of these may be specified using strings for
+        'fromfile', 'tofile', 'fromfiledate', and 'tofiledate'.
+        The modification times are normally expressed in the ISO 8601 format.
+
+        If a or b misses a newline at end of file, an additional newline is
+        generated at end of last line, and an extra warning line "\ No newline
+        at end of file" is generated if warn_no_newline_at_end is True, to
+        make the patch program happy.
+        """
```

Comment:
Nononononono! You didn't copy this from the stdlib, did you? That's not only code duplication but also you're not allowed to do that unless you give them credit and make sure licenses are compatible (which I think they are but still)